ddl/ddl_api.go

```diff
@@ -1815,6 +1815,13 @@ func (d *ddl) CreateTable(ctx sessionctx.Context, s *ast.CreateTableStmt) (err e
 	if err != nil {
 		return errors.Trace(err)
 	}
+	switch s.TemporaryKeyword {
+	case ast.TemporaryGlobal:
+		tbInfo.TempTableType = model.TempTableGlobal
+	case ast.TemporaryLocal:
+		tbInfo.TempTableType = model.TempTableLocal
+	default:
+	}
```

Comment:
It's already checked in `buildTableInfoWithStmt` but it's more complete. E.g. It also checks `EnableGlobalTemporaryTable` and `OnCommitDelete`.
So I think you can extract that part as a function and call it both in `buildTableInfoWithLike` and `buildTableInfoWithStmt`.